src/administration/api-keys.service.ts

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright 2024 IBM Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { FilterQuery, Loaded, ref } from '@mikro-orm/core';
+import dayjs from 'dayjs';
+
+import { ProjectApiKey } from './entities/project-api-key.entity';
+import { ApiKey as ApiKeyDto } from './dtos/api-key';
+import { ApiKeyCreateBody, ApiKeyCreateParams, ApiKeyCreateResponse } from './dtos/api-key-create';
+import { ApiKeyReadParams, ApiKeyReadResponse } from './dtos/api-key-read';
+import { ApiKeyUpdateBody, ApiKeyUpdateParams, ApiKeyUpdateResponse } from './dtos/api-key-update';
+import { ApiKeysListParams, ApiKeysListQuery, ApiKeysListResponse } from './dtos/api-keys-list';
+import { ApiKeyDeleteParams, ApiKeyDeleteResponse } from './dtos/api-key-delete';
+import { getProjectPrincipal } from './helpers';
+import { Project } from './entities/project.entity';
+
+import { createDeleteResponse } from '@/utils/delete';
+import { generateApiKey, scryptApiKey } from '@/auth/utils';
+import { getUpdatedValue } from '@/utils/update';
+import { createPaginatedResponse, getListCursor } from '@/utils/pagination';
+import { ORM } from '@/database';
+import { APIError, APIErrorCode } from '@/errors/error.entity';
+
+export function toDto(apiKey: Loaded<ProjectApiKey>, sensitiveId?: string): ApiKeyDto {
+  return {
+    object: 'organization.project.api_key',
+    id: apiKey.id,
+    name: apiKey.name,
+    created_at: dayjs(apiKey.createdAt).unix(),
+    sensitive_id: typeof sensitiveId === 'string' ? sensitiveId : undefined
```

Comment:
I'd change this to `sensitive_key` or  `secret_key` or something like that. I know it comes from OpenAI but the key is not an ID in our case.